net/tcp/tcp_connect.c

```diff
@@ -359,6 +366,16 @@ int psock_tcp_connect(FAR struct socket *psock,
           ret = psock_setup_callbacks(psock, &state);
           if (ret >= 0)
             {
+              /* Push a cancellation point onto the stack.  This will be
+               * called if the thread is canceled.
+               */
+
+              info.tc_conn = conn;
+              info.tc_cb = state.tc_cb;
+              info.tc_sem = &state.tc_sem;
+              info.tc_free = true;
+              tls_cleanup_push(tls_get_info(), tcp_callback_cleanup, &info);
```

Comment:
Any socket event should be handled accordingly from `devif_callback_s`. If the task is canceled, socket fd should be closed and work queue should be canceled.

Reply:
I just free the callback which thread is been canceled, and I didn't think should close the fd and work queue.Because others thread may using the socket, it is reasonable.